robottelo/utils/sso.py

```diff
@@ -0,0 +1,197 @@
+from robottelo.constants import RHSSO_RESET_PASSWORD
+from robottelo.constants import RHSSO_USER_UPDATE
+from robottelo.datafactory import valid_emails_list
+
+
+class SSOHost(Host):
+    def __init__(self, **kwargs):
+        kwargs['hostname'] = kwargs.get('hostname', settings.rhsso.host_name)
+        super().__init__(**kwargs)
+
+    @lru_cache
+    def get_rhsso_client_id(self, sat_obj):
+        """Getter method for fetching the client id and can be used other functions"""
+        client_name = f'{sat_obj.hostname}-foreman-openidc'
+        self.execute(
+            '{} config credentials '
+            '--server {}/auth '
+            '--realm {} '
+            '--user {} '
+            '--password {}'.format(
+                KEY_CLOAK_CLI,
+                settings.rhsso.host_url.replace('https://', 'http://'),
+                settings.rhsso.realm,
+                settings.rhsso.rhsso_user,
+                settings.rhsso.rhsso_password,
+            ),
+        )
+
+        result = self.execute(f'{KEY_CLOAK_CLI} get clients --fields id,clientId')
+        result_json = json.loads(f'{result}')
+        client_id = None
+        for client in result_json:
+            if client_name in client['clientId']:
+                client_id = client['id']
+                break
+        return client_id
+
+    def get_rhsso_user_details(self, username):
+        """Getter method to receive the user id"""
+        result = self.execute(
+            f"{KEY_CLOAK_CLI} get users -r {settings.rhsso.realm} -q username={username}"
+        )
+        result_json = json.loads(f'[{{{result}')
+        return result_json[0]
+
+    def get_rhsso_groups_details(self, group_name):
+        """Getter method to receive the group id"""
+        result = self.execute(
+            f"{KEY_CLOAK_CLI} get groups -r {settings.rhsso.realm} -q group_name={group_name}"
+        )
+        result_json = json.loads(f'[{{{result}')
+        return result_json[0]
+
+    def upload_rhsso_entity(self, json_content, entity_name):
+        """Helper method upload the entity json request as file on RHSSO Server"""
+        with open(entity_name, "w") as file:
+            json.dump(json_content, file)
+        # self.session.sftp_write(hostname=settings.rhsso.host_name).put(entity_name)
+        self.session.sftp_write(entity_name)
+
+    def create_mapper(self, json_content, client_id):
+        """Helper method to create the RH-SSO Client Mapper"""
+        self.upload_rhsso_entity(json_content, "mapper_file")
+        self.execute(
+            "{} create clients/{}/protocol-mappers/models -r {} -f {}".format(
+                KEY_CLOAK_CLI, client_id, settings.rhsso.realm, "mapper_file"
+            )
+        )
+
+    def create_new_rhsso_user(self, username=None):
+        """create new user in RHSSO instance and set the password"""
+        if not username:
+            username = gen_string('alphanumeric')
+        RHSSO_NEW_USER['username'] = username
```

Comment:
This pattern is copied from the existing module, and should not be continued if you're here to improve the helpers state.

This constant is imported from the module, and then mutated - constants should be immutable (at least treated as such).

Consider creating a basic class, or even just a namedtuple, as an instance attribute. Then it's just tied to this one host, and can be manipulated however is necessary without impacting anything else running in parallel.